.github/workflows/twisted_trunk.yml

```diff
@@ -55,11 +65,24 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Patch dependencies
-        run: .ci/patch_for_twisted_trunk.sh
+        # Note: The script below runs poetry against a virtualenv in /src/.venv/
+        #       ...but the sytest-synapse container expects it to be in /venv/
+        #       So we temporarily symlink it, run poetry, then remove the link.
+        #       Tidying up is important because `/src` is a mounted volume.
+        run: |
+          ln -s -T /venv /src/.venv
+          poetry remove twisted
+          poetry add --extras tls git+https://github.com/twisted/twisted.git#trunk
+          poetry install --no-interaction --extras "all test"
+          rm /src/.venv
         working-directory: /src
       - name: Run SyTest
         run: /bootstrap.sh synapse
         working-directory: /src
+        env:
+          # Use offline mode to avoid reinstalling the pinned version of
+          # twisted.
+          OFFLINE: 1
```

Comment:
I think he twisted trunk CI jobs from today were using the locked environment without this patch?